tools/WebhookProcessor/github_webhook_processor.php

```diff
@@ -0,0 +1,1023 @@
+	global $apiKey;
+	return apisend($url, $method, $content, 'token ' . $apiKey);
+}
+
+function discord_webhook_send($webhook, $content) {
+	return apisend($webhook, 'POST', $content);
+}
+
+function validate_user($payload) {
+	global $validation, $validation_count;
+	$query = array();
+	if (empty($validation))
+		$validation = 'org';
+	switch (strtolower($validation)) {
+		case 'disable':
+			return TRUE;
+		case 'repo':
+			$query['repo'] = $payload['pull_request']['base']['repo']['full_name'];
+			break;
+		default:
+			$query['user'] = $payload['pull_request']['base']['repo']['owner']['login'];
+			break;
+	}
+	$query['author'] = $payload['pull_request']['user']['login'];
+	$query['is'] = 'merged';
+	$querystring = '';
+	foreach($query as $key => $value)
+		$querystring .= ($querystring == '' ? '' : '+') . urlencode($key) . ':' . urlencode($value);
+	$res = github_apisend('https://api.github.com/search/issues?q='.$querystring);
+	$res = json_decode($res, TRUE);
+	return $res['total_count'] >= (int)$validation_count;
+	
+}
+
+function get_labels($payload){
+	$url = $payload['pull_request']['issue_url'] . '/labels';
+	$existing_labels = json_decode(github_apisend($url), true);
+	$existing = array();
+	foreach((array) $existing_labels as $label)
+		$existing[] = $label['name'];
+	return $existing;
+}
+
+function check_tag_and_replace($payload, $title_tag, $label, &$array_to_add_label_to){
+	$title = $payload['pull_request']['title'];
+	if(stripos($title, $title_tag) !== FALSE){
+		$array_to_add_label_to[] = $label;
+		return true;
+	}
+	return false;
+}
+
+function set_labels($payload, $labels, $remove) {
+	$existing = get_labels($payload);
+	$tags = array();
+
+	$tags = array_merge($labels, $existing);
+	$tags = array_unique($tags);
+	if($remove) {
+		$tags = array_diff($tags, $remove);
+	}
+
+	$final = array();
+	foreach($tags as $t)
+		$final[] = $t;
+
+	$url = $payload['pull_request']['issue_url'] . '/labels';
+	echo github_apisend($url, 'PUT', $final);
+}
+
+//rip bs-12
+function tag_pr($payload, $opened) {
+	//get the mergeable state
+	$url = $payload['pull_request']['url'];
+	$payload['pull_request'] = json_decode(github_apisend($url), TRUE);
+	if($payload['pull_request']['mergeable'] == null) {
+		//STILL not ready. Give it a bit, then try one more time
+		sleep(10);
+		$payload['pull_request'] = json_decode(github_apisend($url), TRUE);
+	}
+	
+	$tags = array();
+	$title = $payload['pull_request']['title'];
+	if($opened) {	//you only have one shot on these ones so as to not annoy maintainers
+		$tags = checkchangelog($payload, false);
+
+		if(strpos(strtolower($title), 'refactor') !== FALSE)
+			$tags[] = 'Refactor';
+		
+		if(strpos(strtolower($title), 'revert') !== FALSE)
+			$tags[] = 'Revert';
+		if(strpos(strtolower($title), 'removes') !== FALSE)
+			$tags[] = 'Removal';
+	}
+
+	$remove = array('Test Merge Candidate');
+
+	$mergeable = $payload['pull_request']['mergeable'];
+	if($mergeable === TRUE)	//only look for the false value
```

Comment:
Here is the check for merge ability

Reply:
Have you actually tested whether this checks for merge conflict or not?

Reply:
That’s not one of the things I tested but DTX is using the same version and it works for him, if you require me to test it I can do so when I have time though

Reply:
![image](https://user-images.githubusercontent.com/25063394/63217603-e1179e80-c140-11e9-8c1b-cbf381c1b16f.png)
Image of check working

Reply:
Were you surprised?

Reply:
?